kaitaistruct.py

```diff
@@ -15,6 +15,18 @@
 #
 __version__ = '0.8'
 
+from RecursionSafe import RecursionSafe
+
+recSafe=RecursionSafe()
+
+@recSafe.wrap
+def repr_generator_for_all_props(self):
+  """Generator to use in own __repr__ functions."""
+  return (
+    "".join(( str(k), "=", repr(getattr(self, k)) ))
+    for k in dir(self)
```

Comment:
i don't know if it's usable in this context, but `vars` is often inferior over `dir` and might make test in the following line unnecessary.

Reply:
`vars` doesn't return properties.